### ISSUE TITLE
Context menu windows 11 duplicated

### ISSUE THREAD

**Issue by modz2014** [M2Team/NanaZip#328]
hi i notice the context menu is duplicted under show more options and the windows 11 context menu so if im on windows 11 it should only show up when i right click but not under show more options

**Comment by modz2014** [comment]
i have made temporary fix to it

**Comment by MouriNaruto** [comment]
It's not duplicated due to some third-party apps like Everything will show the classic context menu in their UI. But I think next version of NanaZip will try to add an option for achieve your idea.

Kenji Mouri

**Comment by modz2014** [comment]
any update on it don't mind helping

**Comment by MouriNaruto** [comment]
@modz2014 

I will try to integrate that in recent days.

Kenji Mouri

**Comment by modz2014** [comment]
did you want me to create a pull request I don't mind

**Comment by modz2014** [comment]
Co-authored-by: Modz2014 <dawgz1992.jp@gmail.com>

**Comment by MouriNaruto** [comment]
> is this what you want sorry I'm a bit confused

It can help me to create a commit about those changes co-authored with you.

Kenji Mouri

**Comment by modz2014** [comment]
Ok is that all you need

**Comment by modz2014** [comment]
@MouriNaruto is there going to be new updates to this soon

**Comment by modz2014** [comment]
@MouriNaruto can i send you some archive formats to be added i tried to add them myself but the porject is all the place for me to understand i can email them to you

**Comment by MouriNaruto** [comment]
> can i send you some archive formats to be added i tried to add them myself but the porject is all the place for me to understand i can email them to you

Of course, you can.

Kenji Mouri

**Comment by modz2014** [comment]
ok what is your email address

**Comment by MouriNaruto** [comment]
> ok what is your email address

KurikoMouri@outlook.jp

Kenji Mouri

**Comment by modz2014** [comment]
@MouriNaruto i sent they now sorry i had to finish up

**Comment by modz2014** [comment]
@MouriNaruto I'm trying to add Inno Setup scripts, but it seems a lot harder are you working on adding this as well as other installer

**Comment by modz2014** [comment]
@MouriNaruto ill create a pull request with some changes in the next hour or something for you

**Comment by MouriNaruto** [comment]
I won't accept the suggestion and the PR of hiding NanaZip item in the legacy context menu because there is no proper way to implement.

Due to packaged shell extensions running out of process, we cannot check the progress is File Explorer or third-party applications which need shell extensions.

Due to many people may like to select all files in the folder which the file count is large like the System32 folder, it's not suitable for using okToBeSlow to optimize.


Kenji Mouri

**Comment by modz2014** [comment]
@MouriNaruto wait im confused do you mean about the context menu showing up the new windows 11 one stay hidden under show more options

**Comment by MouriNaruto** [comment]
> do you mean about the context menu showing up the new windows 11 one stay hidden under show more options

Yes, the common open and save dialog only uses the legacy context menu.

Kenji Mouri

**Comment by modz2014** [comment]
can you screenshot to me because it works fine for me

**Comment by MouriNaruto** [comment]
> can you screenshot to me because it works fine for me

![image](https://github.com/M2Team/NanaZip/assets/10867563/8ffd99a5-b5ef-4370-8f15-016260c97ce5)

The simplest way to show what I said is using the Browse dialog when you use Windows Run dialog.

Kenji Mouri

**Comment by MouriNaruto** [comment]
![image](https://github.com/M2Team/NanaZip/assets/10867563/eeaaa16d-71dc-414c-99b8-a6eaf4f9f33b)

Also, when you open File Explorer via Windows Tools in the Start menu, you will always use legacy context menu.

Kenji Mouri

**Comment by modz2014** [comment]
ok i added in checks for this it should work now I will update it later

**Comment by modz2014** [comment]
<img width="881" alt="Untitled" src="https://github.com/M2Team/NanaZip/assets/25472227/e5a0245c-a95a-43a3-8fc0-b7416ff5837a">
also works for browse under run as well

**Comment by modz2014** [comment]
@MouriNaruto Please explain

**Comment by MouriNaruto** [comment]
> @MouriNaruto Please explain

Due to the out-of-process shell extension is hosted by comhost.exe, you cannot get the proper Window which is using the shell extension properly.

Also, your new code snippet is not complete. Not only the common file open and save dialog. I don't think it's good for adding lots of check for that. 

Oh, I forgot something about some users may need to use right click with shift key to show the legacy context menu in File Explorer. So, this proposal is not suitable to implement.

Kenji Mouri

**Comment by MouriNaruto** [comment]
Also, following the Microsoft's current behavior, this is another reason of your proposal is not suitable to implement.

Kenji Mouri

**Comment by modz2014** [comment]
Ok I don't see many users doing that anyway in open and save diag log
because anyway i have one more approach that could fix this isuess
completely but would take some time with it i wont be able to for a couple
of weeks

Also what do you recon how giving the user the option of open the folder
after extarction did you take a look at that

I have half finished of creating iso is something you would be interested
in?

**Comment by modz2014** [comment]
@MouriNaruto i would have to rewrite the context menu and add 2 checks in source code

**Comment by modz2014** [comment]
Hey @MouriNaruto, I've embarked on enhancing the shell extension by transitioning it into a COM Shell Extension. Currently, this transformation is pivotal in overcoming various limitations, such as the recurrent appearance of duplicated context menus in both the new and old interfaces upon right-clicking. Additionally, I've identified and addressed the constraints on subcommands, eliminating any predefined limits and enhancing overall flexibility.

Admittedly, I encountered some complexity in deciphering certain portions of the code shared earlier, but I'm committed to resolving that in the upcoming days. If this development aligns with your interests, I would greatly appreciate your input and support. Let me know, and I'll continue the comprehensive rewrite of the entire codebase.

**Comment by MouriNaruto** [comment]
@modz2014 

>  I've embarked on enhancing the shell extension by transitioning it into a COM Shell Extension. Currently, this transformation is pivotal in overcoming various limitations,

The current implementation is already COM-based. Also, the modern context menu in Windows 11 doesn't support old shell extension interfaces.

> this development aligns with your interests

At least, the "the recurrent appearance of duplicated context menus in both the new and old interfaces upon right-clicking" does not align my interests. The reasons I have already noted before. (keep the same behavior from Microsoft and other third-party implementations, demand from users which needs the legacy menu experience and third-party application interoperability support, reduce the implementation logic)

Kenji Mouri

**Comment by MouriNaruto** [comment]
Also, I don't think it's OK for me to merge your other implementations unless you split them to separate PRs. (I hope each PR has only one feature in the most cases.)

Kenji Mouri

**Comment by modz2014** [comment]
so did you want me to create another pull request for an open extraction folder after extraction is completed?

**Comment by MouriNaruto** [comment]
> so did you want me to create another pull request for an open extraction folder after extraction is completed?

Yes, this is what I hope. Also, I think the modification should be done in NanaZipWindows project or NanaZipG.exe because the GUI extraction implementation is in that module.

Kenji Mouri

**Comment by MouriNaruto** [comment]
About the context menu implementation, I think I will reduce the check logics and menu items for improving the performance.

Kenji Mouri

**Comment by MouriNaruto** [comment]
> OK i will cancel the pull request and reupload them again and i was looking and Notepad++ duplicate of the context menu and they have resolved the issue so I might take a look at this

They use separate COM interfaces for context menus shown in the modern one and the classic one. I don't think it's a good solution because I want to reduce the complexity. I think I will continue to research for the suggestion you requested. And I think I will set co-author-by with you if I commit this feature.

Kenji Mouri

**Comment by modz2014** [comment]
ok no worries i making a commit for Enable Destination Option after extraction under the settings tab will do a pull request soon for it

**Comment by modz2014** [comment]
@MouriNaruto can you link to the extraction part please sorry just trying to work from my phone

**Comment by MouriNaruto** [comment]
> @MouriNaruto can you link to the extraction part please sorry just trying to work from my phone

Maybe you can find something from https://github.com/M2Team/NanaZip/tree/main/SevenZip/CPP/7zip/UI/GUI.

Kenji Mouri

**Comment by modz2014** [comment]
hey what has changed lately I'm back from my holiday today and just trying to figure out where I was

**Comment by modz2014** [comment]
@MouriNaruto i cant see to find the context menu when you right click in NanaZip gui one a file

**Comment by modz2014** [comment]
@MouriNaruto i havent not update NanaZip yet or anything

**Comment by toshiya14** [comment]
I appreciate you have support the new context menu for Windows 11. But actually most of the people hate the new context menu such as me.
I have patched my explorer to roll back classic context menu as default option. In this case, I could not use nanazip function anymore.
As I right click in the explorer, the context menu popped up is classic context menu, and no nanazip options inside.
I guess they might be added into the new context menu, but that menu already be skipped.
So, may I ask you to add an option to switch use classic context menu or new context menu for Windows 11.

Thanks so much.

**Comment by modz2014** [comment]
its still has the old option as well

**Comment by DevX-Cipher** [comment]
@MouriNaruto i have fixed the pain with the out of process you were going  on about a few years back now
